### PR TITLE
enhancement(vrl): add VM option for conditions

### PR DESCRIPTION
**Pull request by StephenWakely**
Ref #11686 

This adds an option to update the condition system to use the VRL Vm.

To start with the default is going to be to use the VM to see if we can get an idea of numbers for the soaks. That will break a large number of tests until #11722 is merged..

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
There seems like a fair bit of duplication here, but I'm not too worried about it since the tree-walking interpreter implementation will go away in the nearish future.

**Comment by StephenWakely** [comment]
>  I'm not too worried about it since the tree-walking interpreter implementation will go away in the nearish future.

This was my thought process too..

**Comment by pablosichert** [comment]
> I'm not too worried about it since the tree-walking interpreter implementation will go away in the nearish future.

On the other hand, we might still have a new runtime mode, LLVM 😄 

Could make this change generic over the runtime as in https://github.com/vectordotdev/vector/pull/11836, but I don't see a problem in deferring since that's not super high impact.

File: src/conditions/vrl.rs [hunk 1]
```diff
@@ -12,6 +14,9 @@ use crate::{
 #[derive(Deserialize, Serialize, Debug, Default, Clone, PartialEq)]
 pub struct VrlConfig {
     pub(crate) source: String,
+
+    #[serde(default)]
```
Comment (pablosichert) [inline src/conditions/vrl.rs:18]: Looking at the soaks: does this currently mean `::Vm`? 🤔

I thought we hadn't flipped over the default yet, and should align with that before merging.

**Review by pablosichert — APPROVED** [review]
Looks good otherwise!

File: src/conditions/vrl.rs (hunk 1 above)
Comment (jszwedko) [reply src/conditions/vrl.rs:18]: See the latest soak tests which have no difference.

Comment (StephenWakely) [reply src/conditions/vrl.rs:18]: Yeah, it's back to Ast now.

Comment (pablosichert) [reply src/conditions/vrl.rs:18]: Oh. I'm blind 😆

**Pull request merged**